phpBB/phpbb/textformatter/s9e/parser.php

```diff
@@ -0,0 +1,355 @@
+	*/
+	public function get_parser()
+	{
+		return $this->parser;
+	}
+
+	/**
+	* {@inheritdoc}
+	*/
+	public function set_var($name, $value)
+	{
+		if ($name === 'max_smilies')
+		{
+			$this->parser->setTagLimit('E', $value ?: PHP_INT_MAX);
+		}
+		else if ($name === 'max_urls')
+		{
+			$this->parser->setTagLimit('URL', $value ?: PHP_INT_MAX);
+		}
+		else
+		{
+			$this->parser->registeredVars[$name] = $value;
+		}
+	}
+
+	/**
+	* {@inheritdoc}
+	*/
+	public function set_vars(array $vars)
+	{
+		foreach ($vars as $name => $value)
+		{
+			$this->set_var($name, $value);
+		}
+	}
+
+	/**
+	* Filter a flash object's height
+	*
+	* @see bbcode_firstpass::bbcode_flash()
+	*
+	* @param  string  $height
+	* @param  integer $max_height
+	* @param  Logger  $logger
+	* @return mixed              Original value if valid, FALSE otherwise
+	*/
+	static public function filter_flash_height($height, $max_height, Logger $logger)
+	{
+		if ($max_height && $height > $max_height)
+		{
+			$logger->err('MAX_FLASH_HEIGHT_EXCEEDED', array('max_height' => $max_height));
+
+			return false;
+		}
+
+		return $height;
+	}
+
+	/**
+	* Filter a flash object's width
+	*
+	* @see bbcode_firstpass::bbcode_flash()
+	*
+	* @param  string  $width
+	* @param  integer $max_width
+	* @param  Logger  $logger
+	* @return mixed              Original value if valid, FALSE otherwise
+	*/
+	static public function filter_flash_width($width, $max_width, Logger $logger)
+	{
+		if ($max_width && $width > $max_width)
+		{
+			$logger->err('MAX_FLASH_WIDTH_EXCEEDED', array('max_width' => $max_width));
+
+			return false;
+		}
+
+		return $width;
+	}
+
+	/**
+	* Filter the value used in a [size] BBCode
+	*
+	* @see bbcode_firstpass::bbcode_size()
+	*
+	* @param  string  $size     Original size
+	* @param  integer $max_size Maximum allowed size
+	* @param  Logger  $logger
+	* @return mixed             Original value if valid, FALSE otherwise
+	*/
+	static public function filter_font_size($size, $max_size, Logger $logger)
+	{
+		if ($max_size && $size > $max_size)
+		{
+			$logger->err('MAX_FONT_SIZE_EXCEEDED', array('max_size' => $max_size));
+
+			return false;
+		}
+
+		if ($size < 1)
+		{
+			return false;
+		}
+
+		return $size;
+	}
+
+	/**
+	* Filter an image's URL to enforce restrictions on its dimensions
+	*
+	* @see bbcode_firstpass::bbcode_img()
+	*
+	* @param  string  $url        Original URL
+	* @param  array   $url_config Config used by the URL filter
+	* @param  Logger  $logger
+	* @param  integer $max_height Maximum height allowed
+	* @param  integer $max_width  Maximum width allowed
+	* @return string|bool         Original value if valid, FALSE otherwise
+	*/
+	static public function filter_img_url($url, array $url_config, Logger $logger, $max_height, $max_width)
+	{
+		// Validate the URL
+		$url = BuiltInFilters::filterUrl($url, $url_config, $logger);
+
+		if ($url === false)
+		{
+			return false;
+		}
+
+		if ($max_height || $max_width)
+		{
+			$stats = @getimagesize($url);
```

Comment:
You will have to use https://github.com/phpbb/phpbb/pull/3382 when it will be merged